Adding Example project with solidity-coverage lib (#1767)
hashgraph/hedera-json-rpc-relay: tools/solidity-coverage-example/README.md

# Sample Hedera/Hardhat/Code coverage Project

> This project demonstrates a basic usage of the code coverage [plugin](https://github.com/sc-forks/solidity-coverage) of HardHat in a Hedera environment. It comes with a sample contract, a test for that contract, and a script that deploys that contract.

Reviewer: ```suggestion
This project demonstrates a basic usage of the solidity coverage [plugin](https://github.com/sc-forks/solidity-coverage) of HardHat in a Hedera environment. Example comes with simple contract to showcase this functionality, as well as more complex example contract with Hedera Precompile system calls. 
```

## The integration

You can find a npm script that launches a http server to serve the code coverage report.
But first the reports have to be generated:
```shell
// to run the tests and generate the reports
npm run test

> // serve the reports on http://127.0.0.1:8080 by default

Reviewer: Why do I need to start a server, just to check the coverage?
Won't a simple `npx hardhat coverage` be enough, just to show the coverage ?

Author: The served report has more detailed data for each file.
But that was a good question. I mentioned the stock command in the README as well (with example output)

# Sample Hardhat Project

> This project demonstrates a basic Hardhat use case with Hedera. It comes with a sample contract, a test for that contract, and a script that deploys that contract.

Reviewer 2: The goal here is to show off solidity-coverage not hardhat, so you should update this description

Try running some of the following tasks:

```shell
npx hardhat help
npx hardhat test

> REPORT_GAS=true npx hardhat test

Reviewer 2: Q: what's this for?

Author: It's part of the default **README.md** generated by the hardhat initial setup. I'll remove it.

npx hardhat node
npx hardhat run scripts/deploy.ts
```

## Hedera and Hardhat integration

You can find a npm script that launches a http server to serve the code coverage report.
But first the reports have to be generated:
```shell
// to run the tests and generate the reports
npm run test
// serve the reports on http://127.0.0.1:8080 by default

> npm run coverage:serve

Reviewer 2: Can you add a print out or image of what a a successful run looks like maybe for the `Lock.sol`

Author: Done

```

### Limitations:

> There is a limitation around hardhat using our precompiled contracts to a precompiled address.

Reviewer 2: ```suggestion
There is a limitation around hardhat using our system contracts at their reserved addresses.
```
